### ISSUE TITLE
How do I use this?

### ISSUE THREAD

**Issue by juju4666** [Perpixel/UE4_QuakeImportBsp#1]
Could you please provide instructions? This looks super cool, but I can't figure out how to add it to Unreal to import Quake maps.

**Comment by Perpixel** [comment]
First, you need to unpack your pak0.pak and pak1.pak. You can use any tool you want for this but I provided my own pakutil with this if you want to use that. Next place the palette.lmp file from pak0.pak in the content folder.

After that you can just import .bsp files and mdl directly from the import button in your content browser.

**Comment by juju4666** [comment]
> 
> 
> First, you need to unpack your pak0.pak and pak1.pak. You can use any tool you want for this but I provided my own pakutil with this if you want to use that. Next place the palette.lmp file from pak0.pak in the content folder.
> 
> After that you can just import .bsp files and mdl directly from the import button in your content browser.

Hi, thank you so much for replying! So, my issue is in getting this to actually work with Unreal. When I add it in to the Plugins folder and try to open my project, Unreal says this plugin has been built with a different engine version (even though I'm on 4.26 and tried on 4.25). I've tried rebuilding the project from source; it compiles but doesn't seem to fix the problem. I haven't been able to find a way to open Unreal with the plugin. Can you help me out?

**Comment by motorsep** [comment]
I tried compiling it with UE4 fork (4.26.2) and I get this error:

E:/UE4Source/UE-oculus-4262-release-1610/Engine/Plugins/Editor/UE4_QuakeImport/Source/QuakeImportBsp/Public/AliasFrameDesc.h(15) : LogCompile: Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
1>E:/UE4Source/UE-oculus-4262-release-1610/Engine/Plugins/Editor/UE4_QuakeImport/Source/QuakeImportBsp/Public/AliasFrameDesc.h(18) : LogCompile: Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
1>E:/UE4Source/UE-oculus-4262-release-1610/Engine/Plugins/Editor/UE4_QuakeImport/Source/QuakeImportBsp/Public/AliasFrameDesc.h(21) : LogCompile: Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
1>E:/UE4Source/UE-oculus-4262-release-1610/Engine/Plugins/Editor/UE4_QuakeImport/Source/QuakeImportBsp/Public/AliasFrameDesc.h(24) : LogCompile: Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
1>E:/UE4Source/UE-oculus-4262-release-1610/Engine/Plugins/Editor/UE4_QuakeImport/Source/QuakeImportBsp/Public/AliasFrameDesc.h(27) : LogCompile: Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.MakeFile.Targets(46,5): error MSB3073: The command "..\..\Build\BatchFiles\Build.bat -Target="UE4Editor Win64 Development" -Target="ShaderCompileWorker Win64 Development -Quiet" -WaitMutex -FromMsBuild" exited with code 6.
1>Done building project "UE4.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 2 up-to-date, 0 skipped ==========

I put UE4_QuakeImport into Plugins/Editor/ and I use MSVC 2019 Community

**Comment by Perpixel** [comment]
It's not an engine plugin. It goes in the plugin folder of your project.

**Comment by juju4666** [comment]
> 
> 
> It's not an engine plugin. It goes in the plugin folder of your project.

We need to put in the source code as well? If so, where does it go?
Could you provide a compiled version for easier access?

**Comment by Perpixel** [comment]
No, sorry, I won't publish binaries. You need to place the entire GitHub depot in a folder inside your project plugins folder. I really just made this for fun and for advanced users.

**Comment by motorsep** [comment]
> No, sorry, I won't publish binaries. You need to place the entire GitHub depot in a folder inside your project plugins folder. I really just made this for fun and for advanced users.

I placed it into Plugins/ in my Project's folder and when UE4 tried building it, I got this error:

![image](https://user-images.githubusercontent.com/1927918/125871321-2bbea2dc-6cc1-4d84-abdf-195eb0a19621.png)

I am using source build, not launcher build of UE4. What can I tweak to make it compile ?

**Comment by juju4666** [comment]
> 
> 
> No, sorry, I won't publish binaries. You need to place the entire GitHub depot in a folder inside your project plugins folder. I really just made this for fun and for advanced users.

Thanks anyway, Perpixel. Your comment helped me to get it working. Awesome tool!

> 
> 
> > No, sorry, I won't publish binaries. You need to place the entire GitHub depot in a folder inside your project plugins folder. I really just made this for fun and for advanced users.
> 
> I placed it into Plugins/ in my Project's folder and when UE4 tried building it, I got this error:
> 
> ![image](https://user-images.githubusercontent.com/1927918/125871321-2bbea2dc-6cc1-4d84-abdf-195eb0a19621.png)
> 
> I am using source build, not launcher build of UE4. What can I tweak to make it compile ?

Here's what you need to do.

1. Create a new project and select C++ (not Blueprint) when creating it
2. Close Visual Studio and Unreal
3.  Add to it a "Plugins" folder and unzip the plugin folder in there.
4. Extract Quake's pak0.pak and pak1.pak somewhere and copy the palette file to the content folder INSIDE plugins/ue4_quakeimporter
5. Try to open Unreal. It'll say it needs to rebuild the plugin, then fail.
6. Right-click the project file (projectname.uproject) and click on Generate Visual Studio project files.
7. A new file will appear, along with a few folders. Open up projectname.sln on Visual Studio
8. Right-click your project name on the right-hand side of VS and click on Compile. It should take a while, then compile successfully.
9. Finally, save and close VS and open up your project. It should now open without issue.
10. Now you just import assets from Quake. The plugin appears to create new folders to store these assets. Some assets failed and crashed UE4 here, like map E1M2, but otherwise it works very well.

**Comment by motorsep** [comment]
@juju4666 ugh, what a hassle :( I guess I'll just stick with the way I currently do this - import BSP into Blender, delete what I need to delete, export as FBX, import into UE4. No crashes!

**Comment by juju4666** [comment]
> 
> 
> @juju4666 ugh, what a hassle :( I guess I'll just stick with the way I currently do this - import BSP into Blender, delete what I need to delete, export as FBX, import into UE4. No crashes!

Well, I mean, you can always use a more complete tool like HammUEr if you integrate BSP into your workflow. This is more of a fun thing to play with. Super cool in that it gets all models, textures and converts to Unreal-friendly formats and applies them back to the map.

**Comment by Perpixel** [comment]
I did this to play quake directly in UE4 so it's not a tool. In theory you have everything you need to remake the game beside the gameplay code. I have a private depot with a bunch of monster working so it is doable.
